cpp/include/cugraph/algorithms.hpp

```diff
@@ -1752,17 +1760,18 @@ std::tuple<rmm::device_uvector<vertex_t>,
            std::optional<rmm::device_uvector<edge_t>>,
            std::optional<rmm::device_uvector<edge_type_t>>,
            rmm::device_uvector<int32_t>,
-           std::optional<rmm::device_uvector<int32_t>>>
+           std::optional<rmm::device_uvector<int32_t>>,
+           std::optional<rmm::device_vector<edge_t>>>
```

Comment:
Won't `edge_t` overflow? `starting_vertices.size()` can possibly overflow 32 bit integers (this may not happen frequently, but we cannot compromise on correctness).

Reply:
For the use cases we've been discussing, I don't think that would happen.  However, as you described in our slack discussions, we want to have the API support future cases as well.  `size_t` would make more sense, I can make that change.

Reply:
Yeah, I believe this should be general purpose, and I hope we have more users in the future.